components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultRequestCoordinator.java

```diff
@@ -476,6 +487,32 @@ && isStepHasMultiOption(context)
         }
     }
 
+    private boolean checkIfApplicationAccessEnabled(HttpServletRequest request, CommonAuthResponseWrapper responseWrapper,
+                                                 AuthenticationContext context) throws IOException, FrameworkException {
+
+        String type = request.getParameter(TYPE);
+        String clientId = request.getParameter(CLIENT_ID_IN_REQUEST);
+
+        // Validate parameters
+        if (StringUtils.isBlank(type)) {
+            type = context.getRequestType();
+        }
+        if (StringUtils.isBlank(clientId)) {
+            clientId = context.getRelyingParty();
+        }
+        if (StringUtils.isNotBlank(type) && StringUtils.isNotBlank(clientId)) {
```

Comment:
This could introduce a potential loophole where we would allow a disabled apps. Check the underlying methods whether they are null safe if so we can ignore the blank checks here and check whether a SP is found. If an SP is not found lets throw an error and break the flow.